tests/everflow/everflow_test_utilities.py

```diff
@@ -247,18 +239,209 @@ def get_port_info(in_port_list, out_port_list, out_port_ptf_id_list, out_port_la
+                                                          for k, v in t1_mg_facts["minigraph_ptf_indices"].items()
+                                                          if k in t1_mg_facts["minigraph_ports"]})
+    setup_information['intf_to_namespace_map'] = {k: mg_facts["minigraph_neighbors"][k]['namespace']
+                                                  for k, v in mg_facts["minigraph_ptf_indices"].items()
+                                                  if k in mg_facts["minigraph_ports"]}
+    setup_information['intf_to_namespace_map'].update({k: t1_mg_facts["minigraph_neighbors"][k]['namespace']
+                                                       for k, v in t1_mg_facts["minigraph_ptf_indices"].items()
+                                                       if k in t1_mg_facts["minigraph_ports"]})
+
+    downstream_block = {
+        'everflow_dut': t3_duthost,
+        'remote_dut': t1_duthost,
+        "router_mac": t3_duthost.facts["router_mac"],
+        "src_port": t3_info['interfaces'][0],
+        "src_port_lag_name": t3_info['pc_names'][0],
+        "src_port_ptf_id": str(t3_info['mgfacts']["minigraph_ptf_indices"][t3_info['interfaces'][0]]),
+        "dest_port": t1_info['interfaces'],
+        "dest_port_ptf_id": t1_info['portids'],
+        "dest_port_lag_name": t1_info['pc_names'],
+    }
+    upstream_block = {
+        'everflow_dut': t1_duthost,
+        'remote_dut': t3_duthost,
+        "router_mac": t1_duthost.facts["router_mac"],
+        "src_port": t1_info['interfaces'][-1],
+        "src_port_lag_name": t1_info['pc_names'][-1],
+        "src_port_ptf_id": str(t1_info['mgfacts']["minigraph_ptf_indices"][t1_info['interfaces'][0]]),
+        "dest_port": list(reversed(t3_info['interfaces'])),
+        "dest_port_ptf_id": list(reversed(t3_info['portids'])),
+        "dest_port_lag_name": list(reversed(t3_info['pc_names'])),
+    }
+    setup_information[UP_STREAM].update(upstream_block)
+    setup_information[DOWN_STREAM].update(downstream_block)
+
+    return setup_information
+
+
+@pytest.fixture(scope="module")
+def setup_info(duthosts, rand_one_dut_hostname, tbinfo, request):
+    """
+    Gather all required test information.
+
+    Args:
+        duthost: DUT fixture
+        tbinfo: tbinfo fixture
+
+    Returns:
+        dict: Required test information
+
+    """
+    topo = tbinfo['topo']['name']
+
+    if 't2' in topo:
+        setup_information = gen_t2_setup_information(duthosts, tbinfo)
+    else:
+        setup_information = gen_t1t0_setup_information(duthosts, rand_one_dut_hostname, tbinfo)
+        duthost = duthosts[rand_one_dut_hostname]
+
     # Disable BGP so that we don't keep on bouncing back mirror packets
     # If we send TTL=1 packet we don't need this but in multi-asic TTL > 1
-    duthost.command("sudo config bgp shutdown all")
+
+    if 't2' in topo:
+        for duthost in duthosts.frontend_nodes:
+            duthost.command("mkdir -p {}".format(DUT_RUN_DIR))
```

Comment:
Is it necessary to put `duthost.command("mkdir -p {}".format(DUT_RUN_DIR))` in the for loop?

Reply:
Can be changed to [setup_information['t3_duthost'], setup_information['t1_duthost']]